File: InventoryManagementServices/ReactApp/inventory-management-react/src/app/models/gameSearch.ts
export interface GameSearch
{
    title: string;
    description: string;
    releaseYear: string;
    gameRatingsId: string;
    gameTypeIds?: [];
}


export class GameSearch implements GameSearch {
    constructor(init?: GameSearchFormValues) {
        Object.assign(this, init);
    }
}

  export class GameSearchFormValues {
    title: string = '';
    description: string= '';
    releaseYear: string= '';
    gameRatingsId: string= '';
    gameTypeIds?: [] = undefined;

    constructor(game?: GameSearchFormValues) {
      if (game) {
        this.title = game.title;
        this.description = game.description;
        this.releaseYear = game.releaseYear;
        this.gameRatingsId = game.gameRatingsId;
        this.gameTypeIds = game.gameTypeIds;
      }
    }
  }

File: InventoryManagementServices/Services/Game/Game.Infrastructure/Entities/GameSearch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Game.Infrastructure.Entities
{
    public class GameSearch
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public int? ReleaseYear { get; set; }
        public Guid? GameRatingsId { get; set; }
        public List<Guid> GameTypeIds { get; set; }
    }
}
